### DISCUSSION TITLE
how to redirect on server error ?

### DISCUSSION THREAD

**Discussion by HananoshikaYomaru** [remix-run/remix#10003] [Q&A]
let me describe what I want to do. I have a specific type of error (called it `ZError`) which could thrown deep inside the code, so sometimes it is uncaught. I want to handle it in the outer most layer. If it is `ZError` , redirect the user to a specific page. How can I achieve something like this without wrapping every loader with a try catch?

**Comment by rururux** [comment]
How about using a wrapper function that wraps the process that might throw a ZError with a try...catch, and if a ZError is caught, redirects accordingly?
Instead of calling the process that might throw the error directly within the loader, I think you can write it more simply by passing arguments or executing it through a wrapper function.
